kf_lib_data_ingest/app/cli.py

```diff
@@ -57,21 +57,27 @@ def common_args_options(func):
         type=click.Path(exists=True, file_okay=True, dir_okay=True),
     )(func)
 
+    # Clear the UID cache
+    func = click.option(
+        "--clear_cache",
+        default=False,
+        is_flag=True,
+        help="Clear the identifier cache before loading into the target service.",
+    )(func)
```

Comment:
Say, I have two batches of data, one already ingested and the other new. 

- When set, does the ingest library remove an existing UID cache and send HTTP requests to the server to 1) GET the first batch's UIDs so it wouldn't create new resources and 2) POST the second batch resources?
- When unset, does the ingest library only send POST requests for the second batch resources?

Reply:
> When this flag is set, does the ingest library remove an existing UID cache

Yes.

> and send HTTP requests to the server to 1) GET the first batch's UIDs

Yes, but only once per entity because it also repopulates the cache with the response results.

> so it wouldn't create new resources

I assume you mean for ones that overlap with the first batch. In theory, yes.

> and 2) POST the second batch resources?

Yes.

---

> When unset, does the ingest library only send POST requests for the second batch resources?

It will send post requests for anything in the current batch because like always the non-key information could have changed.

Reply:
Clearly explained! I am also down for the idea of wiping out an existing cache for every run. This way, we can conisder the data in FHIR as a source of truth, not heavily relying a cache (somethimes deprecated).